// telegram-format/dist/source/markdown.js
"use strict";
// https://core.telegram.org/bots/api#markdown-style
Object.defineProperty(exports, "__esModule", { value: true });
function escape(text) {
    return text.replace(/[*_`[\]()]/g, '');
}
function bold(text) {
    return `*${text.replace(/\*/g, '')}*`;
}
function italic(text) {
    return `_${text.replace(/_/g, '')}_`;
}
function monospace(text) {
    return '`' + text.replace(/`/g, '') + '`';
}
function monospaceBlock(text, programmingLanguage) {
    let result = '';
    result += '```';
    if (programmingLanguage) {
        result += programmingLanguage;
    }
    result += '\n';
    result += text.replace(/```/g, '');
    result += '\n';
    result += '```';
    return result;
}
function url(label, url) {
    return `[${label.replace(/\]/, '')}](${url.replace(/\)/, '')})`;
}
function userMention(label, userId) {
    return url(label, `tg://user?id=${userId}`);
}
exports.markdown = {
    parse_mode: 'Markdown',
    escape,
    bold,
    italic,
    monospace,
    monospaceBlock,
    url,
    userMention
};
